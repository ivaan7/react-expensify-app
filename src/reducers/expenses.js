//expenses reducer

const expensesReducerDefaultState = [];

const expensesReducer = (state = expensesReducerDefaultState, action) => {
    switch (action.type) {
        case "ADD_EXPENSE":
            return [
                ...state,
                action.expense
            ]
        case "REMOVE_EXPENSE":
            return state.filter(({ id }) => {
                return id !== action.id;
            });
        case "EDIT_EXPENSE":
            return state.map((expense) => {
                if (expense.id === action.id) {
                    //if id op expense exists we are taking that expense
                    //and usping spread operator overiding initial values with
                    //one provided with mathod call
                    return {
                        ...expense,
                        ...action.updates
                    };
                } else {
                    return expense;
                }
            });
        case "SET_EXPENSES":
         return action.expenses;
        default:
            return state;
    }
};

export default expensesReducer;