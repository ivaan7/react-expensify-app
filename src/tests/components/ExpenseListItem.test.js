import React from "react";
import {shallow} from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";

test("should reder list item with data",()=>{
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
})
