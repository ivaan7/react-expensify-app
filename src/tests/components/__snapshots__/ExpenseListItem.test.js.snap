// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should reder list item with data 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    gum
  </Link>
  195
   
  0
</div>
`;
